vault/barrier_view.go

```diff
@@ -34,6 +34,10 @@ func NewBarrierView(barrier BarrierStorage, prefix string) *BarrierView {
 	}
 }
 
+func (v *BarrierView) markReadOnly(readOnly bool) {
+	v.readonly = readOnly
```

Comment:
Wondering if we should use a different error message for the read only state from being a replication secondary and the read only state that's due to still setting up

Reply:
I like this idea. It would probably mean that markReadOnly needs to take in a bool and also an error, and store that error as the current one to use if returning a readonly error.

Reply:
I think we can take in just one error parameter instead of both a bool and error. Then remove `readonly` from BarrierView. We could rely on `readOnlyErr` not being nil in the checks.

Reply:
Done.